problems/basic.go

```diff
@@ -66,13 +66,21 @@ func (problem BasicProblem) Error() string {
 	return fmt.Sprintf("%s: %s", problem.ProblemTitle(), problem.Detail)
 }
 
-func (problem *BasicProblem) DecorateWithRequest(ctx context.Context, r *http.Request) {
+func (problem BasicProblem) DecorateWithRequest(ctx context.Context, r *http.Request) Problem {
+	result := BasicProblem{
```

Comment:
Do we need to copy here? I thought just editing the `problem` struct and returning it would work?